### PR TITLE
Validations and Assocations & Color Scheme changes

### PR DESCRIPTION
**Pull request by lhelmert**

### CODE REVIEW

File: app/assets/stylesheets/config/_colors.scss [hunk 1]
```diff
@@ -9,10 +9,15 @@ $light-gray: #dfdfdf;
 //dark black-gray for fonts:
 $black-gray: #3c3c3c;
 //secondary grays for fonts and other elements:
-other_gray_light:#cec4c2;
-other_gray_dark: #a39594;
-
-
+$other_gray_light:#cec4c2;
```
Comment (michaelwlaschitz) [inline app/assets/stylesheets/config/_colors.scss]: other_gray confusing, why not gray_light? or some other characteristic of the grey

File: app/models/ngo.rb [hunk 2]
```diff
@@ -1,4 +1,11 @@
 class Ngo < ApplicationRecord
   belongs_to :user
+  has_many :projects
+  has_many :bookings, through: :projects
   has_one_attached :photo
```
Comment (michaelwlaschitz) [inline app/models/ngo.rb]: delete through Migration

File: app/models/ngo.rb [hunk 3]
```diff
@@ -1,4 +1,11 @@
...
+  has_one_attached :logo
+
+  validates :name, presence: true, uniqueness: true
+  validates :phone, presence: true, uniqueness: true
+  validates :description, presence: true, uniqueness: true
```
Comment (michaelwlaschitz) [inline app/models/ngo.rb]: delete uniqueness

File: app/models/project.rb [hunk 4]
```diff
@@ -1,4 +1,13 @@
 class Project < ApplicationRecord
-  belongs_to :ngo
+  has_one :ngo
```
Comment (michaelwlaschitz) [inline app/models/project.rb]: change to belongs_to :ngo

File: app/models/project.rb [hunk 5]
```diff
@@ -1,4 +1,13 @@
...
+  has_many :bookings
   has_many_attached :photos
+
+  validates :title, length: { maximum: 25 }, uniqueness: true, presence: true
```
Comment (michaelwlaschitz) [inline app/models/project.rb]: delete uniqueness

File: app/models/user.rb [hunk 6]
```diff
@@ -3,5 +3,18 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
+  has_many :reviews
+  has_one :ngo
```
Comment (michaelwlaschitz) [inline app/models/user.rb]: has_one exists? Can he not have more?

File: app/models/user.rb [hunk 7]
```diff
@@ -3,5 +3,18 @@ class User < ApplicationRecord
...
+  has_many :team_memberships
+  has_many :teams, through: :team_memberships
```
Comment (michaelwlaschitz) [inline app/models/user.rb]: belongs_to team

File: app/models/user.rb [hunk 8]
```diff
@@ -3,5 +3,18 @@ class User < ApplicationRecord
...
+  has_many :bookings, through: :teams
+  has_many :ngo_bookings, through: :ngo, source: :bookings
```
Comment (michaelwlaschitz) [inline app/models/user.rb]: NGo booking? Not sure about that one, but might be right

File: app/models/user.rb [hunk 9]
```diff
@@ -3,5 +3,18 @@ class User < ApplicationRecord
...
   has_one_attached :photo
+
+  validates :first_name, presence: true
+  validates :last_name, presence: true
+  validates :email, format: { with: /\A.*@.*\.com\z/ }, presence: true, uniqueness: true
+  validates :password, presence: true
+  validates :description, presence: true
+  validates :phone, presence: true, uniqueness: true, uniqueness: true
```
Comment (michaelwlaschitz) [inline app/models/user.rb]: twice uniqueness, delete one

File: app/assets/stylesheets/config/_colors.scss (hunk 1 above)
Comment (lhelmert) [reply app/assets/stylesheets/config/_colors.scss]: secondary_grey_light
secondary_grey_dark

File: app/models/ngo.rb (hunk 2 above)
Comment (lhelmert) [reply app/models/ngo.rb]: delete has_one_attached :photo

**Pull request closed without merging**